collector/src/main/java/com/bitmovin/analytics/data/SimpleEventDataDispatcher.java

```diff
@@ -1,53 +1,61 @@
 package com.bitmovin.analytics.data;
 
+import android.content.Context;
+
 import com.bitmovin.analytics.BitmovinAnalyticsConfig;
+import com.bitmovin.analytics.license.Licenser;
 import com.bitmovin.analytics.utils.DataSerializer;
 import com.bitmovin.analytics.utils.HttpClient;
-import com.bitmovin.analytics.utils.LicenseCall;
-import com.bitmovin.analytics.utils.LicenseCallback;
+import com.bitmovin.analytics.license.LicenseCall;
+import com.bitmovin.analytics.license.OnAuthCompleted;
 
 import java.util.Iterator;
 import java.util.Queue;
 import java.util.concurrent.ConcurrentLinkedQueue;
 
-public class SimpleEventDataDispatcher implements IEventDataDispatcher, LicenseCallback {
+public class SimpleEventDataDispatcher implements IEventDataDispatcher, OnAuthCompleted {
     private static final String TAG = "SimpleDispatcher";
+    private final Licenser licenser;
 
     private Queue<EventData> data;
     private HttpClient httpClient;
     private boolean enabled = false;
     private BitmovinAnalyticsConfig config;
-    private LicenseCallback callback;
+    private OnAuthCompleted callback;
+    private Context context;
 
     private int sampleSequenceNumber = 0;
 
-    public SimpleEventDataDispatcher(BitmovinAnalyticsConfig config, LicenseCallback callback) {
+    public SimpleEventDataDispatcher(BitmovinAnalyticsConfig config, Context context, OnAuthCompleted callback, Licenser licenser) {
+        this.licenser = licenser;
         this.data = new ConcurrentLinkedQueue<EventData>();
-        this.httpClient = new HttpClient(config.getContext(), BitmovinAnalyticsConfig.analyticsUrl);
+        this.httpClient = new HttpClient(context, config.getAnalyticsUrl());
         this.config = config;
         this.callback = callback;
+        this.context = context;
     }
 
     @Override
-    synchronized public void authenticationCompleted(boolean success) {
+    synchronized public void authenticationCompleted(boolean success, String key) {
         if (success) {
```

Comment:
This was obviously missing before but now it will happen more often: We need to clear the queue and not add new things to the queue. Otherwise this is a memory leak